geometry/rotation.hpp

```diff
@@ -53,6 +89,92 @@ class Rotation : public LinearMap<FromFrame, ToFrame> {
            Multivector<double, ToFrame, rank_y> y_from_frame,
            Multivector<double, ToFrame, rank_z> z_from_frame);
 
+  // While expressing one basis in another is unambiguous to define passive
+  // rotations, rotations expressed from an axis and an angle are more
+  // confusing, and Euler angles are a constant trap.  We therefore require a
+  // tag that specifies which frame is being defined by the given angle and
+  // axis.
+  // These constructors come in pairs (with either value of DefinesFrame), like
+  // the matrix constructors above.  For each pair, one of the constructors has
+  // an additional |typename = void|, see above.
+
+  template<typename Scalar,
+           typename F = FromFrame,
+           typename T = ToFrame,
+           typename = std::enable_if_t<!std::is_same<F, T>::value>>
+  Rotation(Angle const& angle,
```

Comment:
I realize that I suggested the tag type, but I think that it would be a bit nicer/more readable to use the "to frame" template parameter to document that you are defining the frame.  I'd like to be able to write:
```
geometry::Rotation<ICRFJ2000Equator, DefineFrame<ICRFJ2000Ecliptic>>(
  23 * Degree + 26 * ArcMinute + 21.448 * ArcSecond,
  geometry::Bivector<double, ICRFJ2000Equator>({-1, 0, 0}));
```
So basically, enable if the "to frame" parameter is a `DefineFrame`.